pype/plugins/global/publish/extract_review.py

```diff
@@ -13,21 +16,1085 @@ class ExtractReview(pyblish.api.InstancePlugin):
+
+        self.log.debug("Input path {}".format(full_input_path))
+        self.log.debug("Output path {}".format(full_output_path))
+
+    def audio_args(self, instance, temp_data):
+        """Prepares FFMpeg arguments for audio inputs."""
+        audio_in_args = []
+        audio_filters = []
+        audio_out_args = []
+        audio_inputs = instance.data.get("audio")
+        if not audio_inputs:
+            return audio_in_args, audio_filters, audio_out_args
+
+        for audio in audio_inputs:
+            # NOTE modified, always was expected "frameStartFtrack" which is
+            # STRANGE?!!! There should be different key, right?
+            # TODO use different frame start!
+            offset_seconds = 0
+            frame_start_ftrack = instance.data.get("frameStartFtrack")
+            if frame_start_ftrack is not None:
+                offset_frames = frame_start_ftrack - audio["offset"]
+                offset_seconds = offset_frames / temp_data["fps"]
+
+            if offset_seconds > 0:
+                audio_in_args.append(
+                    "-ss {}".format(offset_seconds)
+                )
+            elif offset_seconds < 0:
+                audio_in_args.append(
+                    "-itsoffset {}".format(abs(offset_seconds))
+                )
+
+            audio_in_args.append("-i \"{}\"".format(audio["filename"]))
+
+        # NOTE: These were changed from input to output arguments.
+        # NOTE: value in "-ac" was hardcoded to 2, changed to audio inputs len.
+        # Need to merge audio if there are more than 1 input.
+        if len(audio_inputs) > 1:
+            audio_out_args.append("-filter_complex amerge")
+            audio_out_args.append("-ac {}".format(len(audio_inputs)))
+
+        return audio_in_args, audio_filters, audio_out_args
+
+    def rescaling_filters(self, temp_data, output_def, new_repre):
+        """Prepare vieo filters based on tags in new representation.
+
+        It is possible to add letterboxes to output video or rescale to
+        different resolution.
+
+        During this preparation "resolutionWidth" and "resolutionHeight" are
+        set to new representation.
+        """
+        filters = []
+
+        letter_box = output_def.get("letter_box")
+
+        # Get instance data
+        pixel_aspect = temp_data["pixel_aspect"]
+
+        # NOTE Skipped using instance's resolution
+        full_input_path_single_file = temp_data["full_input_path_single_file"]
+        input_data = pype.lib.ffprobe_streams(full_input_path_single_file)[0]
+        input_width = input_data["width"]
+        input_height = input_data["height"]
+
+        self.log.debug("pixel_aspect: `{}`".format(pixel_aspect))
+        self.log.debug("input_width: `{}`".format(input_width))
+        self.log.debug("resolution_height: `{}`".format(input_height))
+
+        # NOTE Setting only one of `width` or `heigth` is not allowed
+        output_width = output_def.get("width")
```

Comment:
The reason to do it this way is that both width and height must be set to avoid using width from `output_def` and height from `instance.data`. Edge case but possible...

Reply:
I see. Well the code I have posted is simplified version of your original code so perhaps you have in it some broken logic.

Reply:
```
output_def = {
    "width": 1920
}

temp_data = {
     "resolution_width": 800,
     "resolution_height": 600
}

output_width = output_def.get("width") or temp_data["resolution_width"] or input_width
output_height = output_def.get("height") or temp_data["resolution_height"] or input_height

# In this case the result of simplified version will be
output_width = 1920
output_height = 600
```

Reply:
yes that is correct, you were right ;)